rust/noosphere-sphere/src/petname/write.rs

```diff
@@ -24,17 +24,23 @@ where
     K: KeyMaterial + Clone + 'static,
     S: Storage + 'static,
 {
-    /// Configure a petname, by assigning some [Did] to it or none. By assigning none,
-    /// the petname is implicitly removed from the address space (note: this does not
-    /// erase the name from historical versions of the sphere). If a name is set that
-    /// already exists, the previous name shall be overwritten by the new one, and any
-    /// associated [Jwt] shall be unset.
+    /// Configure a petname, by assigning some [Did] to it or none. By assigning
+    /// none, the petname is implicitly removed from the address space (note:
+    /// this does not erase the name from historical versions of the sphere). If
+    /// a name is set that already exists, the previous name shall be
+    /// overwritten by the new one, and any associated [Jwt] shall be unset.
     async fn set_petname(&mut self, name: &str, identity: Option<Did>) -> Result<()>;
 
-    /// Configure a petname, assigning some [Did] to it and setting its
-    /// associated [Jwt] to a known value. The [Jwt] must be a valid UCAN that
-    /// publishes a name record and grants sufficient authority from the
-    /// configured [Did] to the publisher.
+    /// Set the [LinkRecord] associated with a petname.  The [LinkRecord] must
+    /// resolve a valid UCAN that authorizes the corresponding sphere to be
+    /// published and grants sufficient authority from the configured [Did] to
+    /// the publisher. The audience of the UCAN must match the [Did] that was
+    /// most recently assigned the associated petname. Note that a petname
+    /// _must_ be assigned to the audience [Did] in order for the record to be
+    /// set.
+    async fn set_petname_record(&mut self, name: &str, record: &LinkRecord) -> Result<Option<Did>>;
```

Comment:
Might be worth mentioning WHY we can't just implement `set_petname_record()` as previously implemented, rather than a two-step process of `set_petname` + `set_petname_record` e.g. documenting the need for this change

Reply:
You mean as a comment? Can you paint a picture of who that benefits?

Reply:
Me in 6 months thinking, "why doesn't `set_petname_record()` also just set a petname->identity mapping in one-go?", e.g. why it's mandatory that a petname must already be assigned before adoption, though thinking about it more, this is similar to rolling other verifications from the NS worker into the petname writers

Reply:
I'll add a code comment reminder :+1: